### PR TITLE
Support for sycl reducers in RAJA launch

### PR DESCRIPTION
**Pull request by artv3**
This PR adds sycl reducer support to RAJA::launch 

Interesting error when using more than 151 work-items: 
terminate called after throwing an instance of 'sycl::_V1::exception'
  what():  The implementation handling parallel_for with reduction requires work group size not bigger than 151

### CODE REVIEW

File: examples/launch-param-reductions.cpp [hunk 1]
```diff
@@ -50,10 +53,25 @@ int main(int RAJA_UNUSED_ARG(argc), char** RAJA_UNUSED_ARG(argv[]))
 //
   constexpr int N = 1000000;
 
+//
```
Comment (rhornung67) [inline examples/launch-param-reductions.cpp:56]: Thought: would the example in this file be more useful if there was a method templated on resource, exec policy, etc. that contained a generic reduction kernel and is called for each back-end in a specialized way?  That would simplify and reduce the amount of code and emphasize RAJA portability.

Comment (artv3) [reply examples/launch-param-reductions.cpp:56]: I think that would be nice for all our examples. One challenge though is that resources aren't templated. We would still have to setup of the #ifdef for each type. If we wanted a generic function that contained a reduction kernel we could template on policy and pass in a typed erased the resource (still requiring the ifdefs). Then we can just call the reduction kernel function with our different backends. 

Perhaps we can discuss some ideas with a subset of the team on this.

Comment (rhornung67) [reply examples/launch-param-reductions.cpp:56]: Sounds good. This idea  shouldn't hold up your PR. Something to work through when we have spare time.

**Review by rhornung67 — APPROVED** [review]

File: examples/launch-param-reductions.cpp (hunk 1 above)
Comment (MrBurmark) [reply examples/launch-param-reductions.cpp:56]: Take a look at what I've been trying with a `for_each_tuple` function in the multi reducer PR https://github.com/LLNL/RAJA/blob/c0b1e1b8fd4e50830bd842c19db2769d0498b21d/examples/forall_multi-reductions.cpp

File: examples/launch-param-reductions.cpp [hunk 2]
```diff
@@ -39,6 +37,11 @@ constexpr int CUDA_BLOCK_SIZE = 256;
 constexpr int HIP_BLOCK_SIZE = 256;
 #endif
 
+#if defined(RAJA_ENABLE_SYCL)
+//LC testing hardware has a limit of 151
+constexpr int SYCL_BLOCK_SIZE = 151;
```
Comment (MrBurmark) [inline examples/launch-param-reductions.cpp:42]: Maybe 128 as it seems nicer than 151?

Comment (rhornung67) [reply examples/launch-param-reductions.cpp:42]: I was going to suggest that also.

Comment (artv3) [inline examples/launch-param-reductions.cpp:42]:
```suggestion
constexpr int SYCL_BLOCK_SIZE = 128;
```

File: examples/launch-param-reductions.cpp (hunk 1 above)
Comment (MrBurmark) [reply examples/launch-param-reductions.cpp:56]: I made an old mistake in that example, I declared a host device lambda in a generic lambda which is not allowed in cuda.

**Review by rchen20 — APPROVED** [review]

**Pull request merged**